### DISCUSSION TITLE
How can I calculate the number of standard cells in a window

### DISCUSSION THREAD

**Discussion by b224hisl** [The-OpenROAD-Project/OpenROAD#4880] [General]
Excuse me, I would like the get the number of std cells of at a certain window. That what I have done in the gui:
1. First only the std cells can  be selected:
![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/39954153/0cccbee1-7da6-4e14-988d-d26051926123)

2. Hightlight some cells:
![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/39954153/966bb051-0ab6-46b3-9ba9-9bbae72d69c2)
![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/39954153/d97c7dc0-786a-40ad-8781-8ba3948e0773)

3. Then I can get the number through this:
![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/39954153/e1dd22b0-1ceb-49da-a513-84f4ef826cdc)
it's 332

But how can I make it by writing a script to set a range of coordinates and get the number of std cells. I found there is a 'select' command. I think what a gui can do should be done by a script. Can anyone give me a hand?

**Comment by maliberty** [comment]
There is a select command but it doesn't have a "by area" option currently.  Having selected you can use gui::select_next to iterate through.

The simplest is to just walk the db directly.  Something like:
```
set count 0
set block [ord::get_db_block]
foreach inst [$block getInsts] {
  set bbox [$inst getBBox]
  if { [$bbox xMin] >= 100 && [$bbox xMax] <= 10000 &&
       [$bbox yMin] >= 100 && [$bbox yMax] <= 10000} {
    incr count
  }
}

puts "Found $count"
```

**Reply by maliberty** [reply]
Note the units will be DBU not microns

**Reply by b224hisl** [reply]
OK fine， thank you very much~
